pkg/endpointmanager/manager.go

```diff
@@ -405,12 +405,20 @@ func AddEndpoint(owner endpoint.Owner, ep *endpoint.Endpoint, reason string) (er
 	}
 
 	if build {
-		go func() {
+		if asynchronouslyBuild {
+			go func() {
+				if err := ep.RegenerateWait(owner, reason); err != nil {
+					ep.Logger("endpointmanager").Error("removing endpoint because initial build failed")
+					Remove(ep)
```

Comment:
It seems odd to me to regenerate an endpoint and if something fails we remove it from the endpointmanager without having the caller of this function to be aware of that.

Reply:
Yeah, I was a bit unsure about what to do here, so I erred on the side of caution by keeping the existing behavior of removing the endpoint from the EndpointManager if the first build for it failed :/